From c9eb72b35b8722ff6c72d139b41a39730fff0593 Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Wed, 9 Oct 2024 11:18:50 +0530
Subject: [PATCH] Add IPMB Support Port patch to Nuvoton kernel source to
 ensure IPMB uses I2C protocol over SMBus.

Ref: https://git.ami.com/core/ami-bmc/base-tech/os.linux.kernel.openbmc.linux/-/commit/e1b0bafb048cefdd4b2fb403512e3faae0a2a093

Tested and Verified:
1. Successfully able to get IPMB Request
root@intel-obmc:~# busctl call xyz.openbmc_project.Ipmi.Channel.Ipmb \
/xyz/openbmc_project/Ipmi/Channel/Ipmb org.openbmc.Ipmb \
sendRequest yyyyay 1 6 0 1 0
(iyyyyay) 0 7 0 1 0 15 32 129 13 6 2 191 0 0 0 2 2 70 22 3 0
root@intel-obmc:~#

Signed-off-by: vipinc <vipinc@ami.com>
---
 drivers/char/ipmi/ipmb_dev_int.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/char/ipmi/ipmb_dev_int.c b/drivers/char/ipmi/ipmb_dev_int.c
index a0e9e80d92ee..ec21f190fe71 100644
--- a/drivers/char/ipmi/ipmb_dev_int.c
+++ b/drivers/char/ipmi/ipmb_dev_int.c
@@ -326,10 +326,12 @@ static int ipmb_probe(struct i2c_client *client)
 	ret = misc_register(&ipmb_dev->miscdev);
 	if (ret)
 		return ret;
-
+#if 1 /* FIXME: Quick fix. Need to add parsing code for ipmb_id instead */
+	ipmb_dev->is_i2c_protocol = true;
+#else
 	ipmb_dev->is_i2c_protocol
 		= device_property_read_bool(&client->dev, "i2c-protocol");
-
+#endif
 	ipmb_dev->client = client;
 	i2c_set_clientdata(client, ipmb_dev);
 	ret = i2c_slave_register(client, ipmb_slave_cb);
-- 
2.25.1

